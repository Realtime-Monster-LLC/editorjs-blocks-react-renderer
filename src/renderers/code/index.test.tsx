import React from 'react';
import { create } from 'react-test-renderer';
import Code, { CodeBlockData } from '.';

describe('<Code />', () => {
  describe('when receives a code block', () => {
    const data: CodeBlockData = {
      code: 'body {\n font-size: 14px;\n line-height: 16px;\n}',
      lang: 'text/html',
    };

    it('renders a <pre> tag with <code>', () => {
      expect(create(<Code data={data} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('when receives a code block with className', () => {
    const data: CodeBlockData = {
      code: 'body {\n font-size: 14px;\n line-height: 16px;\n}',
      lang: 'text/html',
    };

    it('renders a <pre> tag with <code className>', () => {
      expect(create(<Code data={data} className={"code-block"} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('when receives a code block without lang', () => {
    const data: CodeBlockData = {
      code: 'body {\n font-size: 14px;\n line-height: 16px;\n}',
    };

    it('renders a <pre> tag with <code> without lang property', () => {
      expect(create(<Code data={data} />).toJSON()).toMatchSnapshot();
    });
  });
});
