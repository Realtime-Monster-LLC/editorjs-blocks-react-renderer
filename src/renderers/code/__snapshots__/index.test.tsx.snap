// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code /> when receives a code block renders a <pre> tag with <code> 1`] = `
<pre>
  <code
    lang="text/html"
  >
    body {
 font-size: 14px;
 line-height: 16px;
}
  </code>
</pre>
`;
